src/Media/MediaPlayer.cpp

```diff
@@ -713,6 +837,25 @@ void MPlayer::HouseMovieLoop() {
     }
 }
 
+std::shared_ptr<IMovie> MPlayer::loadFullScreenMovie(std::string_view movieFileName) {
+    Blob blob = LoadMovie(movieFileName);
+    if (!blob) {
+        return nullptr;
+    }
+
+    std::shared_ptr<Movie> pMovie = std::make_shared<Movie>();
+    if (!pMovie->LoadFromLOD(blob)) {
+        return nullptr;
+    }
+
+    bool setupSuccess = pMovie->prepare();
+    if (!setupSuccess) {
+        return nullptr;
+    }
+
+    return std::dynamic_pointer_cast<IMovie>(pMovie);
```

Comment:
Just `return pMovie;` doesn't work here?

Reply:
Ah well, that's a good question. I kinda kept that code as is from the original place but I can give it a try.

Reply:
I just changed that `pMovie` into a simple `unique_ptr` since there's no sharing part involved when using `VideoState`.